pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,44 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it won't stop until a cancel signal happens.
+func (p *Puller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) error {
+	if err := retry.Do(ctx, p.retry, func(ctx context.Context) error {
+		req := &loggingpb.TailLogEntriesRequest{
+			ResourceNames: []string{p.resource},
+			Filter:        filter,
+		}
+		var tailLogEntriesClient loggingpb.LoggingServiceV2_TailLogEntriesClient
+		var createClientErr error
+		if tailLogEntriesClient, createClientErr = p.client.TailLogEntries(ctx); createClientErr != nil {
+			return fmt.Errorf("failed to create tailLogEntriesClient: %w", createClientErr)
+		}
+		defer func() {
+			_ = tailLogEntriesClient.CloseSend()
+		}()
```

Comment:
This should be outside of retry, otherwise each retry creates a new client.

Reply:
Moved to outside of retry.